contributors/design-proposals/arbitrator.md

```diff
@@ -0,0 +1,265 @@
+  to each application, i.e. if all applications wanted to use all free resources, then they would be allowed to do so 
+  in some relative proportion)
+* If some application A is using less than its guarantee, and then if it decides to use its guarantee and there aren't 
+  enough free resources to do so, it should be able to evict tasks from some other application or applications
+  (that is/are using more than their guarantee) in order to obtain its guarantee
+
+Further, group "bigdata" apps and "service" apps into two buckets, providing each bucket (in aggregate) guaranteed 
+access to some fraction of the cluster, and best-effort access to the entire cluster with the understanding that usage 
+above the guarantee can be revoked at any time. 
+
+According to this [mesos-style.md](https://github.com/kubernetes/kubernetes/blob/master/docs/devel/mesos-style.md), 
+it definitely can build a customized components to do so; but it’s better for Kubernetes to provide such a general feature
+ about resource planning, resource management and resource sharing.
+
+## Terminology
+
+* Arbitrator: A new components/binary that allocate resources based on policy (resource allocation).
+  The default scheduler still assign Pods to resources (resource assignment) 
+* Deserved (Resource): The total number of resources that the arbitrator allocated to the namespace
+* Overused: The namespace is overused if it used more resources than deserved
+
+## Context
+
+As Kubernetes growing up, there’s several features on resource QoS and sharing between namespaces.
+
+### Preemption and Re-scheduler:
+
+A pod can be evicted because some other pod needs the resources it is using (preemption). There’s a priority-based preemption 
+scheme (each pod has a priority, and pods with higher and possibly equal priority can preempt it; who makes the decision of which
+pods to preempt is TBD but could be the default scheduler, the rescheduler, and/or application-specific controllers 
+with integrated scheduling functionality). Preemption always uses graceful termination. A priority scheme generally implies that
+quota is allocated on a per-priority-level basis, so that applications can be given a limited amount of quota at the highest priority level
+and a much larger amount of quota (perhaps even infinite, i.e. the cluster's entire capacity) at lower priority level(s).
+And rescheduler evicts pods to enforce cluster-level policies (currently there’s a proto-rescheduler that enforces
+the policy "critical pods like Heapster, DNS, etc. should never be blocked from running due to insufficient free resources
+in the cluster" but there are many other policies it could enforce). It works by evicting one or more pods to
+allow some pending pod(s) to schedule.
+
+The preemption is required to shuffle resources between namespaces; arbitrator will be one of the decision maker who defined “priority”,
```

Comment:
Is this assuming that namespaces can `exceed` their quota or quota has not be set for a namespace?